docs/Go-with-VS-Code-FAQ-and-Troubleshooting.md

```diff
@@ -85,9 +85,29 @@ _Tip_: If you are looking for logs after doing a particular operation, first cle
      - If you see expected results in the terminal, but not in VS Code, log an issue in the [vscode-go](https://github.com/Microsoft/vscode-go) repo, else 
 log an issue in the [gocode](https://github.com/mdempsky/gocode) repo. If you are using Go modules, log the issue in https://github.com/stamblerre/gocode
 
-**Q: Why doesn't formatting doesn't work on file save?**
-
-**A:** Check the logs (View -> Output -> Log (Extension Host) from the drop down in the top right corner) for messages like "Formatting took too long" or "Format On Save feature could be aborted". If you find such a message, then chances are formatted was aborted because it took too long and so can affect the save experience. You can control this timeout using the setting `editor.formatOnSaveTimeout`
+**Q: File saving is sometimes slow or stuck running save participants.**
+
+**A:** You may noticed this problem because of the window popup with a
+message such as "Running Save Participants ...". This indicates the
+underlying tools used for file formatting or other code action such as
+package imports take longer than desired. Before 1.42, VS Code aborted
+the code action and skipped formatting, which resulted in file saving
+without expected formatting or import. After 1.42, VS Code changed to
+surface the issue and let users either wait or cancel the entire file
+save operation.
+
+A workaround is either 1) disable "On Save" features, or 2) cancel the
```

Comment:
I'd rather not advertise disabling on save stuff? Maybe first we encourage people to file an issue and as a last resort they can disable.

Reply:
PTAL.